components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py

```diff
@@ -127,9 +139,19 @@ def _create_job_request(
 
         request["TransformResources"]["InstanceType"] = inputs.instance_type
         request["TransformResources"]["InstanceCount"] = inputs.instance_count
-        request["TransformResources"]["VolumeKmsKeyId"] = inputs.resource_encryption_key
-        request["DataProcessing"]["InputFilter"] = inputs.input_filter
-        request["DataProcessing"]["OutputFilter"] = inputs.output_filter
+
+        if inputs.resource_encryption_key:
+            request["TransformResources"][
+                "VolumeKmsKeyId"
+            ] = inputs.resource_encryption_key
+        else:
+            request["TransformResources"].pop("VolumeKmsKeyId")
+
+        if inputs.input_filter:
```

Comment:
Looks like all the parameters of data processing are being marked as optional. input_filter, output_filter are now optional and join_source was already optional

if none of the parameters for [data processing](https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/common/templates/transform.template.yaml#L24) are provided, it should be popped?

Can you please add this corresponding logic?

Reply:
Thanks for your review.
I added the logic to check https://github.com/kubeflow/pipelines/pull/8336/commits/1c014590377e1569d029770364ade1d9fb840ecc

Reply:
Hey @ittus, by default the [request[DataProcessing] will not be empty](https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/common/templates/transform.template.yaml#L24-L27) so the following if condition will never be true
```
if not request["DataProcessing"]:
     request.pop("DataProcessing")
```

I made some assumptions in my previous comment based on the fields but after reading [this documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/batch-transform-data-processing.html#batch-transform-data-processing-workflow) I realized that we only need to pop the filters if not specified.

We just need to do:
```
if inputs.output_filter:
    request["DataProcessing"]["OutputFilter"] = inputs.output_filter
else:
    request["DataProcessing"].pop("OutputFilter")
```
same for input filter

Reply:
@surajkota 
Thanks for checking again.
I fixed in https://github.com/kubeflow/pipelines/pull/8336/commits/aa8764ddf598f31a9a5ffcaceb58c3e9247e192b
Please help to recheck 🙇‍♂️